src/main/java/calculator/separator/DefaultSeparator.java

```diff
@@ -0,0 +1,32 @@
+package calculator.separator;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefaultSeparator implements Separator {
+    private final String separator = "[,:]"; //[] 안의 어떤 문자든 일치
+    /*
+    //()그룹
+    \\d 숫자(0-9)
+    ()+ 하나 이상 반복
+     */
+    private final String regex = "(\\d+(" + separator + "))+\\d";
+
+    @Override
+    public List<Long> separate(String userInput) {
+        if (!userInput.matches(regex)) {
+            throw new IllegalArgumentException();
+        }
+        return separateNumbers(userInput);
+    }
+
+    @Override
+    public List<Long> separateNumbers(String userInput) {
+        String numbers = userInput.replaceAll(separator, "");
+        List<Long> nums = new ArrayList<>();
+        for (int i = 0; i < numbers.length(); i++) {
+            nums.add(Long.parseLong(String.valueOf(numbers.charAt(i))));
```

Comment:
코드에 여러 함수가 중첩되어 있어서 가독성 측면에서 세 줄로 나누는건 어떠신지 궁금합니다!

```suggestion
           String stringNumber = String.valueOf(numbers.charAt(i));
           Long number = Long.parseLong(stringNumber);
           nums.add(number);
```

Reply:
> 코드에 여러 함수가 중첩되어 있어서 가독성 측면에서 세 줄로 나누는건 어떠신지 궁금합니다!

오 저 부분이 좀 복잡해보이네요! 다음부턴 나눠보겠습니다!